Update OSCAR banner

paemurru opened the pull request:
In today's OSCAR meeting it was decided to decide by Monday which banner to use. Use the corresponding reaction to vote.

😄 Original OSCAR banner
![vote_1_current](https://github.com/oscar-system/Oscar.jl/assets/143521159/e5939d44-cf97-41f8-a0f0-fdfa346e1386)

&nbsp;
&nbsp;
&nbsp;

🎉 Very slight change
![vote_2_curved](https://github.com/oscar-system/Oscar.jl/assets/143521159/3e6d1f9a-d334-402c-af24-4789ca999481)

&nbsp;
&nbsp;
&nbsp;

🚀 Julia-style, letters have vertical bars like in the current [OSCAR logo](https://avatars.githubusercontent.com/u/25104835?s=200&v=4) (according to the meeting, the current OSCAR logo might change soon)
![vote_3_julia_mimic_oscar_logo](https://github.com/oscar-system/Oscar.jl/assets/143521159/905cb7e1-2e90-47cd-8608-edba109dbea3)

&nbsp;
&nbsp;
&nbsp;

❤️ Julia-style, clean version with no vertical bars
![vote_4_julia_clean](https://github.com/oscar-system/Oscar.jl/assets/143521159/29fc7558-6d3b-4439-a3db-0974db2ebfa8)

paemurru committed:
New banner

paemurru committed:
Raw strings to avoid '\' being treated as escaping

wdecker commented:
Von meinem iPhone gesendetAm 21.02.2024 um 18:37 schrieb Erik Paemurru ***@***.***>:﻿New banner
banner_new.png (view on web)
Old banner
banner_old.png (view on web)



You can view, comment on, or merge this pull request online at:
  https://github.com/oscar-system/Oscar.jl/pull/3410

Commit Summary

  b1a53be  New banner
  fc8e2d4  Raw strings to avoid '\' being treated as escaping


File Changes  (1 file)

  
    M
    src/Oscar.jl
    (14)
  


Patch Links:

  https://github.com/oscar-system/Oscar.jl/pull/3410.patch
  https://github.com/oscar-system/Oscar.jl/pull/3410.diff


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

wdecker commented:
Much better, thanks!Von meinem iPhone gesendetAm 21.02.2024 um 19:03 schrieb Wolfram Decker ***@***.***>:﻿Von meinem iPhone gesendetAm 21.02.2024 um 18:37 schrieb Erik Paemurru ***@***.***>:﻿New banner
banner_new.png (view on web)
Old banner
banner_old.png (view on web)



You can view, comment on, or merge this pull request online at:
  https://github.com/oscar-system/Oscar.jl/pull/3410

Commit Summary

  b1a53be  New banner
  fc8e2d4  Raw strings to avoid '\' being treated as escaping


File Changes  (1 file)

  
    M
    src/Oscar.jl
    (14)
  


Patch Links:

  https://github.com/oscar-system/Oscar.jl/pull/3410.patch
  https://github.com/oscar-system/Oscar.jl/pull/3410.diff


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

fingolfin reviewed (dismissed):
fine by me

paemurru committed:
Merge branch 'oscar-system:master' into EP/Banner

lgoettgens commented:
Please adapt the example output in the README.md file as well.

paemurru commented:
Since we always call OSCAR from Julia, I thought perhaps it is nice to match the OSCAR banner style with Julia's, what do you think?
![banner_figlet](https://github.com/oscar-system/Oscar.jl/assets/143521159/45eb8368-ffbc-4c42-967b-cda8a5fe86cf)

paemurru committed:
figlet style OSCAR banner, update README.md

joschmitt commented:
I would be in favour of option # 3, if this is a vote. (Or anything else that at least partially resembles the "actual" logo from the website.)

thofma commented:
I must have missed the discussion, where we decided that we want to update the banner.

Since other stakeholders might have an opinion, we can quickly discuss it tomorrow.

paemurru committed:
Remove README.md julia syntax color

While it is possible to have Julia syntax highlighting in REPL (terminal
or command line) using for example OhMyREPL, the first-time user will
probably not have this.

paemurru committed:
More similarity to logo from website

paemurru commented:
Added more similarity with the OSCAR logo from the website.
![banner_4_actual_logo](https://github.com/oscar-system/Oscar.jl/assets/143521159/9653786f-bd77-44be-a0e9-7ed791a58749)

aaruni96 commented:
While we are taking the Julia banner as inspiration, it might be a nice idea to also add a link to the Oscar documentation along the Oscar banner.

fingolfin commented:
I don't care much, but let me point out that one issue with the Julia banner is that it doesn't work well if the terminal width is below a certain threshold (there is even a PR for this: <https://github.com/JuliaLang/julia/pull/51811>). While the same holds for our current banner but the threshold is much smaller. 

Don't take this as an objection, just as something to be considered.

fingolfin commented on src/Oscar.jl:
```diff
@@ -39,21 +39,11 @@ if Sys.iswindows()
-  println()
-  println("Type: '?Oscar' for more information")
-  println("(c) 2019-2024 by The OSCAR Development Team")
+  println(raw"  ___  ____   ____    _    _.__       ")
```
Shouldn't the first line also have the `|` if you want to match Julia? And no trailing whitespace?
```suggestion
  println(raw"  ___  ____   ____    _    _.__    |")
```

paemurru replied:
Actually, the first line in the Julia banner just consists of the underscore character! To match it, I have made that the first line in the OSCAR banner consists of only underscore characters

fingolfin commented on src/Oscar.jl:
```diff
@@ -39,21 +39,11 @@ if Sys.iswindows()
+  println(raw" / | \/ ___| / ___|  / \  | |_ \   |  Combining ANTIC, GAP, Polymake, Singular")
+  println(raw"| | | \_|_ \| |     / . \ | |_) |  |  Type \"?Oscar\" for more information")
+  println(raw"| |_| |___) | |___ / /_\ \| |_ <   |  ")
+  println(raw" \_|_/\____/ \____/_/ _ \_\_| \_\  |  " * "Version $VERSION_NUMBER")
```
You could also use a triple quoted string
```suggestion
  println(
    raw"""  ___  ____   ____    _    _.__    |  Combining ANTIC, GAP, Polymake, Singular
           / | \/ ___| / ___|  / \  | |_ \   |  Type "?Oscar" for more information
          | | | \_|_ \| |     / . \ | |_) |  |  Manual: https://docs.oscar-system.org
          | |_| |___) | |___ / /_\ \| |_ <   |
           \_|_/\____/ \____/_/ _ \_\_| \_\  |  Version """ * $VERSION_NUMBER)
```

paemurru committed:
Triple quoted string, manual link, text to top row

Co-authored-by: Max Horn <max@quendi.de>

aaruni96 commented on src/Oscar.jl:
```diff
@@ -39,21 +39,12 @@ if Sys.iswindows()
+           / | \/ ___| / ___|  / \  | |_ \   |  Type "?Oscar" for more information
+          | | | \_|_ \| |     / . \ | |_) |  |  Manual: https://docs.oscar-system.org
+          | |_| |___) | |___ / /_\ \| |_ <   |
+           \_|_/\____/ \____/_/ _ \_\_| \_\  |  Version """ * $VERSION_NUMBER)
```
You have to enclose version number in string

```suggestion
           \_|_/\____/ \____/_/ _ \_\_| \_\  |  Version """ * "$VERSION_NUMBER")
```

paemurru committed:
Fix string, remove text from top line

paemurru commented:
Looks like this now
![banner_5_link](https://github.com/oscar-system/Oscar.jl/assets/143521159/178d2e07-c8b6-4ce0-b6e9-2cd2d06c0dd8)

lgoettgens commented on README.md:
```diff
@@ -31,21 +31,13 @@ in the OSCAR manual to learn more on how to contribute to OSCAR.
-... which comes with absolutely no warranty whatsoever
-Type: '?Oscar' for more information
-(c) 2019-2024 by The OSCAR Development Team
+  ___  ____   ____    _    _.__
```
```suggestion
  ___  ____   ____    _    _ __
```
the single dot in the topmost row looks somehow off, it's too high

joschmitt commented:
If this an ASCII art design competition I want to chuck in:
```
  _____  _____   _____     _     _  ___
 / _|_ \/  ___\ / ____|   / \   | | _  \
| /   \ | (___ | /       / . \  | | _) |
| |   | \__|  \| |      / / \ \ | | _ <
| \___/ |___)  | \____ / / = \ \| |  \ \
 \__|__/\_____/ \_____/_/  =  \_\_|  |_|
```
It's one line higher (and a few characters wider), but I think the S looks better with the extra line.
Okay, back to work for me... :)

paemurru commented:
Together with the rest of the banner text, the design by @joschmitt does not fit in a 80-character width terminal, which is a big down-side

joschmitt commented:
> Together with the rest of the banner text, the design by @joschmitt does not fit in a 80-character width terminal, which is a big down-side

Yeah, I think the "fine-print" can also come below the logo. But yes, if we want to go full julia-style this doesn't work.

paemurru committed:
Remove dot from top row in README.md

Co-authored-by: Lars Göttgens <lars.goettgens@gmail.com>

paemurru committed:
Remove dot from top row in Oscar.jl

paemurru commented:
Removed top row dot character as recommended by @lgoettgens
![banner_6_dot](https://github.com/oscar-system/Oscar.jl/assets/143521159/d6afcc1e-b540-407a-88b7-0db973631d33)

JohnAAbbott commented:
To my eyes the current (old) banner is the worst: all the newer proposals are better!
I understand the reason for wanting the vertical lines in the banner (to mimic those in the logo), but to my eyes they look like "something went wrong" in the ASCII art version.  TBH, it is not that important to me.
Where do we vote?

lgoettgens commented:
> Where do we vote?

@paemurru will create a poll in this thread.

paemurru committed:
Banner without vertical bars inside the letters

paemurru committed:
Merge branch 'oscar-system:master' into EP/Banner

HereAround commented:
> > Where do we vote?
> 
> @paemurru will create a poll in this thread.

Despite reading the corresponding message on slack, I am uncertain how the vote is supposed to work. So...

I like `Julia-style, clean version with no vertical bars` best.

Thank you for working on this!

(I agree with @JohnAAbbott. Also to my eyes, the vertical lines in the third banner look like "something went wrong" in the ASCII art version.)

paemurru commented:
@HereAround Thank you! I would not have noticed myself that the vertical lines look odd.

How to vote: at the end of the first post of this pull request, there is a button to add a reaction.
![scrn1 b](https://github.com/oscar-system/Oscar.jl/assets/143521159/67bafa15-dba3-4a62-af8a-44221911be51)
To vote, put a reaction according to which banner you wish to vote for.
![scrn2 b](https://github.com/oscar-system/Oscar.jl/assets/143521159/55069adb-ff5f-4229-9aef-4ebabb335084)

wdecker commented:
After looking again at the two julia like designs, I believe that the O and/or the distance to the S are too small. Von meinem iPhone gesendetAm 24.02.2024 um 11:32 schrieb Erik Paemurru ***@***.***>:﻿
@HereAround Thank you! I would not have noticed myself that the vertical lines look odd.
How to vote: at the end of the first post of this pull request, there is a button to add a reaction.
scrn1.b.png (view on web)
To vote, put a reaction according to which banner you wish to vote for.
scrn2.b.png (view on web)

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

paemurru committed:
Merge branch 'oscar-system:master' into EP/Banner

Krastanov commented:
Just for reference, a similar question and a lot of banner-handling code is being discussed for julia itself: https://github.com/JuliaLang/julia/pull/51811

In particular, the new StyledStrings standard library can be pretty helpful.

paemurru committed:
Increase distance between O and S

paemurru committed:
Print one-line banner for small terminals

paemurru committed:
Merge branch 'oscar-system:master' into EP/Banner

paemurru commented:
Increased spacing between O and S according to suggestion by @wdecker.
```
  ___   ____   ____    _    ____
 / _ \ / ___| / ___|  / \  |  _ \   |  Combining ANTIC, GAP, Polymake, Singular
| | | |\___ \| |     / _ \ | |_) |  |  Type "?Oscar" for more information
| |_| | ___) | |___ / ___ \|  _ <   |  Manual: https://docs.oscar-system.org
 \___/ |____/ \____/_/   \_\_| \_\  |  Version 1.1.0-DEV
```

Added a one-line banner for small terminals (width less than 79 characters)
```
OSCAR 1.1.0-DEV  https://docs.oscar-system.org  Type "?Oscar" for help
```
